proxy/http/proxy/server.py

```diff
@@ -360,20 +361,8 @@ def generate_upstream_certificate(
         with self.lock:
             if not os.path.isfile(cert_file_path):
                 logger.debug('Generating certificates %s', cert_file_path)
-                # TODO: Parse subject from certificate
-                # Currently we only set CN= field for generated certificates.
-                gen_cert = subprocess.Popen(
-                    ['openssl', 'req', '-new', '-key', self.flags.ca_signing_key_file, '-subj',
-                     f'/C=/ST=/L=/O=/OU=/CN={ text_(self.request.host) }'],
-                    stdout=subprocess.PIPE,
-                    stderr=subprocess.PIPE)
-                sign_cert = subprocess.Popen(
-                    ['openssl', 'x509', '-req', '-days', '365', '-CA', self.flags.ca_cert_file, '-CAkey',
-                     self.flags.ca_key_file, '-set_serial', str(self.uid.int), '-out', cert_file_path],
-                    stdin=gen_cert.stdout,
-                    stderr=subprocess.PIPE)
-                # TODO: Ensure sign_cert success.
-                sign_cert.communicate(timeout=10)
+                pki.gen_web(cert_file_path, self.flags.ca_signing_key_file, self.flags.ca_key_file,
+                            self.flags.ca_cert_file, text_(self.request.host), str(self.uid.int))
```

Comment:
I kind of fail to understand the motive behind all the changes.

a) I think they won't work cross platform e.g. will break on MacOS
b) IIUC, all we wanted was `SAN` extension in the signing step.  So you could have only modified `sign_csr` to incorporate `SAN` extension when called from `server.py`

Reply:
We need SAN and SHA 256. As your link say : 
> SHA-1 signed certificates are no longer trusted for TLS. 

If i remember well, we must have a base cert/key/ca in sha256, to sign in sha256. It's for this reason that I change a big part of the implementation.
I will take a look on it today.